modules/pipeline/providers/clusterinfo/interface.go

```diff
@@ -119,15 +129,27 @@ func (p *provider) GetClusterInfoByName(clusterName string) (apistructs.ClusterI
 	if ok {
 		return cluster, nil
 	}
-	clusterInfo, err := p.bdl.GetCluster(clusterName)
+	var clusterInfo *apistructs.ClusterInfo
+	var err error
+	// first try to get cluster info from cluster-manager, because cluster-manager implemented caching for easy and fast,
+	// if got err and it's current cluster , try to get from k8s configmap
+	clusterInfo, err = p.bdl.GetCluster(clusterName)
```

Comment:
you will always get an error when you try to connect cluster-manager at edge side, it's unneccesary.

check edge deploy structure directly is better.

Reply:
adjusted, if edge deploy and it's current cluster name will try to query from k8s, otherwise query from cluster-namager